docs/topics/testing/advanced.txt

```diff
@@ -67,6 +67,35 @@ The following is a simple unit test using the request factory::
             response = MyView.as_view()(request)
             self.assertEqual(response.status_code, 200)
 
+.. _topics-testing-advanced-allowed-hosts:
+
+Tests and ``ALLOWED_HOSTS``
+===========================
+
+.. versionadded:: 1.11
+
+If your application includes unit tests with custom host names, those tests
```

Comment:
For Django's tests or third-party apps that aren't actually deployed on their own, I agree with the `@override_settings` approach. For the settings file approach I was thinking about projects that are actually meant to serve multiple domains, for example, https://github.com/django/djangoproject.com/.